File: frontend/src/types/document.ts
/**
 * Document TypeScript types for InsightGuide
 * Aligned with backend Pydantic schemas
 */

export type DocumentStatus =
  | 'uploaded'
  | 'processing'
  | 'converted'
  | 'analyzing'
  | 'analyzed'
  | 'failed'

export type FileType = 'pdf' | 'docx' | 'doc' | 'md' | 'txt' | 'generated'

export interface Document {
  id: string
  userId: string
  project_id?: string
  stakeholder_profile_id?: string
  interview_round_id?: string
  guide_version?: number
  is_frozen?: boolean
  title: string
  sourceFileUrl: string
  fileType: FileType
  status: DocumentStatus
  createdAt: string
  updatedAt: string
  costUsd?: number
  aiUsage?: Record<string, unknown>
}

export interface DocumentAnalysis {
  documentId: string
  status: DocumentStatus
  themes: ThemeSummary[]
  questionCardsCount: number
  createdAt: string
  updatedAt: string
  costUsd: number
  aiUsage: Record<string, unknown>
}

export interface ThemeSummary {
  id: string
  documentId: string
  themeNumber: number
  title?: string | null
  rationale?: string | null
  brdMapping?: string[]
  priority?: number
  estimatedMinutes?: number | null
  questionCardsCount?: number
}

File: backend/app/schemas/document.py
"""Document-related Pydantic schemas."""

from datetime import datetime
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field


class DocumentCreate(BaseModel):
    """Schema for creating a new document."""

    title: str = Field(..., min_length=1, max_length=255, description="Document title")


class DocumentResponse(BaseModel):
    """Schema for document response."""

    id: str
    user_id: str
    project_id: Optional[str] = None
    title: str
    source_file_url: str
    file_type: str  # pdf, docx, doc, md, txt
    status: str
    created_at: datetime
    updated_at: datetime
    cost_usd: float = 0.0
    ai_usage: Dict[str, Any] = Field(default_factory=dict)

    class Config:
        from_attributes = True


class DocumentStatus(BaseModel):
    """Schema for document status."""

    id: str
    status: str
    message: Optional[str] = None
    cost_usd: float = 0.0
    ai_usage: Dict[str, Any] = Field(default_factory=dict)


class DocumentAnalysisResponse(BaseModel):
    """Schema for document analysis results."""

    document_id: str
    status: str
    slides: List[dict] = Field(default_factory=list)
    topic_cards_count: int = 0
    created_at: datetime
    updated_at: datetime
    cost_usd: float = 0.0
    ai_usage: Dict[str, Any] = Field(default_factory=dict)
